linux_pipeline/Jenkinsfile_kernel_bisect_controller

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env groovy
+
+String kernelTree;
+String pipelineName;
+
+properties ([
+    [$class: 'ParametersDefinitionProperty',
+        parameterDefinitions: [
+        [$class: 'ChoiceParameterDefinition',
+            choices: 'bisect-commit-validator\nazure-pipeline\nupstream-pipeline\ndeveloper-pipeline',
+            name: 'JOB_TO_RUN',
+            description: 'Job to run bisect on'],
+        [$class: 'ChoiceParameterDefinition',
+            choices: 'linux-next\nazure-kernel\nnet-next',
+            name: 'KERNEL_TREE',
+            description: 'What Kernel tree to be used (Available only for upstream and developer pipelines).'],
+        [$class: 'StringParameterDefinition',
+            name: 'KERNEL_GIT_BRANCH',
+            defaultValue: 'master',
+            description: 'Branch to be built'],
+        [$class: 'StringParameterDefinition',
+            name: 'GOOD_COMMIT',
+            defaultValue: 'ad8009875f8f9d02bef7f2fb59a34b3ad323640b',
+            description: 'Good commit (this goes back)'],
+        [$class: 'StringParameterDefinition',
+            name: 'BAD_COMMIT',
+            defaultValue: '735071426eeb9631b419120f82b261d799ddde69',
+            description: 'Bad commit (can be HEAD)'],
+        [$class: 'StringParameterDefinition',
+            name: 'ENABLED_STAGES',
+            defaultValue: 'build_artifacts, publish_temp_artifacts, boot_test',
+            description: 'What stages to run. By default all stages are enabled']
+        ]
+    ]
+])
+
+node ("bisect_kernel_builder") {
+    resolveGitRepo(JOB_TO_RUN, KERNEL_TREE)
```

Comment:
this is the main logic, you should add comments on what are the steps and what each step does.